Sources/KDTree.swift

```diff
@@ -18,93 +18,121 @@ public enum KDTree<Element: KDTreePoint> {
     indirect case node(left: KDTree<Element>, value: Element, dimension: Int, right: KDTree<Element>)
 }
 
+fileprivate extension UnsafeMutablePointer {
+    @_transparent
+    func swapAt(_ i: Int, _ j: Int) {
+        let temp = self[i]
+        self[i] = self[j]
+        self[j] = temp
+    }
+}
+
 extension KDTree {
     public init(values: [Element], depth: Int = 0) {
         guard !values.isEmpty else {
             self = .leaf
             return
         }
         
-        let currentSplittingDimension = depth % Element.dimensions
-        if values.count == 1, let firstValue = values.first {
-            self = .node(left: .leaf, value: firstValue, dimension: currentSplittingDimension, right: .leaf)
-        }
-        else {
-            var median = values.count / 2
+        let count = values.count
+        
+        let pointer = UnsafeMutablePointer<Element>.allocate(capacity: count)
+        
+        // copy values from the array
+        pointer.initialize(from: values, count: count)
+        
+        self = KDTree(values: pointer, startIndex: 0, endIndex: count)
```

Comment:
oops, this should be
```suggestion
        self = KDTree(values: pointer, startIndex: 0, endIndex: count, depth: depth)
```